test/harry/main/org/apache/cassandra/harry/SchemaSpec.java

```diff
@@ -0,0 +1,440 @@
+            {
+                commaAppender.accept(sb);
+                sb.append(column.name).append(' ').append(column.isReversed() ? "DESC" : "ASC");
+            }
+
+            sb.append(')');
+        }
+
+        return sb.toString();
+    }
+
+    private String getPrimaryKeyCql()
+    {
+        StringBuilder sb = new StringBuilder();
+        sb.append("PRIMARY KEY (");
+        if (partitionKeys.size() > 1)
+        {
+            sb.append('(');
+            SeparatorAppender commaAppender = new SeparatorAppender();
+            for (ColumnSpec<?> cd : partitionKeys)
+            {
+                commaAppender.accept(sb);
+                sb.append(cd.name);
+            }
+            sb.append(')');
+        }
+        else
+        {
+            sb.append(partitionKeys.get(0).name);
+        }
+
+        for (ColumnSpec<?> cd : clusteringKeys)
+            sb.append(", ").append(cd.name);
+
+        return sb.append(')').toString();
+    }
+
+    public String toString()
+    {
+        return String.format("schema {cql=%s}", compile());
+    }
+
+    private static Runnable doOnce(Runnable r)
+    {
+        return new Runnable()
+        {
+            boolean executed = false;
+
+            public void run()
+            {
+                if (executed)
+                    return;
+
+                executed = true;
+                r.run();
+            }
+        };
+    }
+
+    public static class SeparatorAppender implements Consumer<StringBuilder>
+    {
+        boolean isFirst = true;
+        private final String separator;
+
+        public SeparatorAppender()
+        {
+            this(",");
+        }
+
+        public SeparatorAppender(String separator)
+        {
+            this.separator = separator;
+        }
+
+        public void accept(StringBuilder stringBuilder)
+        {
+            if (isFirst)
+                isFirst = false;
+            else
+                stringBuilder.append(separator);
+        }
+
+        public void accept(StringBuilder stringBuilder, String s)
+        {
+            accept(stringBuilder);
+            stringBuilder.append(s);
+        }
+
+
+        public void reset()
+        {
+            isFirst = true;
+        }
+    }
+
+    public boolean equals(Object o)
+    {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        SchemaSpec that = (SchemaSpec) o;
+        return Objects.equals(keyspace, that.keyspace) &&
+               Objects.equals(table, that.table) &&
+               Objects.equals(partitionKeys, that.partitionKeys) &&
+               Objects.equals(clusteringKeys, that.clusteringKeys) &&
+               Objects.equals(regularColumns, that.regularColumns);
+    }
+
+    public int hashCode()
+    {
+        return Objects.hash(keyspace, table, partitionKeys, clusteringKeys, regularColumns);
+    }
+
+    public interface Options
+    {
+        boolean addWriteTimestamps();
```

Comment:
this was for accord but it doesn't have to be... so still makes sense to keep...